flink-streaming-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBase.java

```diff
@@ -94,5 +122,9 @@ public void close() {
 		} catch (Exception e) {
 			LOG.error("Error while closing cluster.", e);
 		}
+		Throwable e = exception.get();
+		if (e != null) {
+			LOG.error("Error while sending value.", e);
```

Comment:
This needs to be rethrown, otherwise close() may complete without an exception, but the result may be incomplete.